apps/shipments/views.py

```diff
@@ -22,12 +23,26 @@ class ShipmentViewSet(viewsets.ModelViewSet):
     serializer_class = ShipmentSerializer
     permission_classes = ((permissions.IsAuthenticated, IsOwner) if settings.PROFILES_ENABLED
                           else (permissions.AllowAny,))
+    filter_backends = (filters.SearchFilter, filters.OrderingFilter, DjangoFilterBackend,)
+    searchable_fields = ['mode', 'pickup_estimated', 'delivery_estimated']
+    for location_field in ['ship_from_location', 'ship_to_location', 'final_destination_location']:
+        for field in ['name', 'address_1', 'city', 'state', 'postal_code', 'country']:
+            searchable_fields.append(f'{location_field}__{field}')
+
+    search_fields = tuple(searchable_fields)
+    ordering_fields = ('modified_at', 'created_at', 'pickup_estimated', 'delivery_estimated')
     http_method_names = ['get', 'post', 'delete', 'patch']
 
     def get_queryset(self):
         queryset = self.queryset
         if settings.PROFILES_ENABLED:
             queryset = queryset.filter(owner_id=self.request.user.id)
+        if 'ship_from_location_not_null' in self.request.query_params:
```

Comment:
I believe if you use a FilterSet as described above, you can take out all this not_null stuff, because it should be provided by the FilterSet.

Reply:
I was unable to get it to search for when it was not null using filters, as the search/filter seems to only work for charfields.